app/celery/tasks.py

```diff
@@ -87,6 +88,16 @@
 )
 
 
+@notify_celery.task(name="update-job")
+@statsd(namespace="tasks")
+def update_in_progress_jobs():
+    jobs = dao_get_in_progress_jobs(datetime.utcnow() - timedelta(minutes=5))
```

Comment:
I wonder if that 5 minutes restriction could not bite us later. What if a beat task is skipped, due to redeployment? would that still be ok? What is the reasoning for picking 5 minutes?

Reply:
The idea was to speed things up by only looking in the time window since the last heartbeat, the last time it was called. However in testing this morning I did run into the issue you are describing. I'll remove this for now.

Reply:
Yeah I think we should run this on all "in progress" jobs. Or if we want to get fancy, we could run it on "in progress" jobs where the last update time is more than 10 or 20 minutes ago. That way we would catch the jobs this feature is concerned with - the ones that may be passed to `process_incomplete_jobs`